app/code/Magento/QuoteGraphQl/etc/schema.graphqls

```diff
@@ -368,3 +372,22 @@ type Order {
     order_number: String!
     order_id: String @deprecated(reason: "The order_id field is deprecated, use order_number instead.")
 }
+
+type CheckoutUserInputError @doc(description:"An error encountered while adding an item the the cart.") {
+    message: String! @doc(description: "Localized error message")
+    path: [String]! @doc(description: "Path to the input field that caused an error. See the GraphQL specification about path errors for details: http://spec.graphql.org/draft/#sec-Errors")
+    code: CheckoutUserInputErrorCodes! @doc(description: "Checkout-specific error code")
+}
+
+type AddProductsToCartOutput {
+    cart: Cart!
+    userInputErrors:[CheckoutUserInputError]!
```

Comment:
@rogyar using this userInputErrors:[CheckoutUserInputError], will have an impact on the reorder schema. I am not sure if this should be named with "Checkout" here. 
cc: @paliarush

Reply:
This violates modularity because there is no dependency of Quote on Sales.

Reply:
@paliarush Exactly that was my concern too. I have updated the schema.